### PR TITLE
Introduce a sample for the ComposeWorkflow concept.

### PR DESCRIPTION
**Pull request by zach-klippenstein**
Part of #458.

~This is pretty weird and sketchy, probably don't want to merge, but wanted to get it up for discussion anyway.~

This is only a sample now.

### CODE REVIEW

**Comment by rjrjr** [comment]
What would you think about making this a sample?

**Comment by rjrjr** [comment]
Yeah, I get your ambivalence. I really want to play with this; have no idea if it's a good idea.

Nice thing about making it a sample is that it stays at least a bit visible w/o support being implied. We can loop back and play with it, see if others do.

**Comment by zach-klippenstein** [comment]
I'd be good with making it a sample.

File: samples/compose-samples/src/main/java/com/squareup/sample/compose/hellocomposeworkflow/ComposeWorkflow.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  override fun asStatefulWorkflow(): StatefulWorkflow<PropsT, *, OutputT, ComposeRendering> =
+    ComposeWorkflowImpl(this)
+}
+
+/**
+ * Returns a [ComposeWorkflow] that renders itself using the given [render] function.
+ */
+@WorkflowUiExperimentalApi
+public inline fun <PropsT, OutputT : Any> Workflow.Companion.composed(
```
Comment (steve-the-edwards) [inline samples/compose-samples/src/main/java/com/squareup/sample/compose/hellocomposeworkflow/ComposeWorkflow.kt:51]: What's the use case for this extension?

**Review by steve-the-edwards — APPROVED** [review]

File: samples/compose-samples/src/main/java/com/squareup/sample/compose/hellocomposeworkflow/ComposeWorkflow.kt (hunk 1 above)
Comment (rjrjr) [reply samples/compose-samples/src/main/java/com/squareup/sample/compose/hellocomposeworkflow/ComposeWorkflow.kt:51]: Same as `Workflow.Companion.stateless` and '.stateful`. Lets us create anonymous Workflow instances.

File: samples/compose-samples/src/main/java/com/squareup/sample/compose/hellocomposeworkflow/HelloComposeWorkflow.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          .wrapContentSize(Alignment.Center)
+      )
+    }
+  }
+}
+
+@OptIn(WorkflowUiExperimentalApi::class)
+@Preview(showBackground = true)
+@Composable fun HelloComposeWorkflowPreview() {
+  // TODO(#458) Uncomment once preview support is imported.
```
Comment (rjrjr) [inline samples/compose-samples/src/main/java/com/squareup/sample/compose/hellocomposeworkflow/HelloComposeWorkflow.kt:47]: Didn't you already merge this?

File: samples/compose-samples/src/main/java/com/squareup/sample/compose/hellocomposeworkflow/HelloComposeWorkflowActivity.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    val model: HelloComposeModel by viewModels()
+    setContentView(
+      WorkflowLayout(this).apply {
+        start(
+          renderings = model.renderings,
+          registry = ViewRegistry()
+        )
+      }
```
Comment (rjrjr) [inline samples/compose-samples/src/main/java/com/squareup/sample/compose/hellocomposeworkflow/HelloComposeWorkflowActivity.kt:26]:
```suggestion
      WorkflowLayout(this).apply { start(model.renderings) }
```

**Review by rjrjr — APPROVED** [review]

**Comment by zach-klippenstein** [comment]
Test failure: 
```
com.squareup.sample.compose.launcher.SampleLauncherTest > allSamplesLaunch[test(AVD) - 7.0] FAILED 
	java.lang.AssertionError: Failed to launch sample Inline ComposeRendering
	at com.squareup.sample.compose.launcher.SampleLauncherTest.allSamplesLaunch(SampleLauncherTest.kt:31)
```

It's passing locally on my 29 emulator though, so might be a version-specific thing. Testing on a lower emulator…

**Comment by zach-klippenstein** [comment]
Ah, it's probably because the screen size/density is different and the last sample is now hidden below the viewport. Need to scroll to it in the test.

**Pull request merged**